src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -33,16 +33,12 @@ public NSWGovAdaptor() throws Exception {
   }
 
   private AQDataPoint convertDataPoint(NSWGovAQDataPoint givenPoint) { 
-    // Take the site id from given NSWGovAQDataPoint.
-    int pointSiteid = givenPoint.getId();
+    // Take the site id from given NSWGovAQDataPoint and fetch coordinates from map.
+    int pointSiteid = givenPoint.siteId;
+    Coordinates pointCoord = new Coordinates(this.dataMap.get(pointSiteid).lat, this.dataMap.get(pointSiteid).lng);
```

Comment:
```suggestion
    GovSiteDetails siteDetails = this.dataMap.get(pointSiteid);
    Coordinates pointCoord = new Coordinates(siteDetails.lat, siteDetails.lng);
```

And similarly below; remove the repeated Map lookup, use `siteDetails` instead.